This adds in example code for a CRD.

moorepants opened the pull request:
This example is of a single variate CRD design with a basic ANOVA. In this case, the factor (virus) is significant with respect to predicting the yield.

Things to check on:
- [x] How do transformations fit in with these analyses?
- [x] Add boxplot
- [x] Add Tukey's 1-df test
- [x] Is the Tukey test correct?
- [x] I don't think an interaction plot is necessary (only one factor). The box plot should provide all you need.
- [x] Transform final bar graph into non-ggplot code.

Current Text Output:

```
> source('crd_one_var.R', print.eval = TRUE)
Loading required package: lattice
Loading required package: plyr
Shapiro-Wilk Normality Test
===============================================================================

    Shapiro-Wilk normality test

data:  residuals(model)
W = 0.9588, p-value = 0.7663

===============================================================================
Levene's Test
===============================================================================
Levene's Test for Homogeneity of Variance (center = median)
      Df F value Pr(>F)
group  3  0.4004 0.7566
       8               
===============================================================================
ANOVA Table
===============================================================================
Analysis of Variance Table

Response: yield
          Df  Sum Sq Mean Sq F value    Pr(>F)    
virus      3 1170.21  390.07  17.345 0.0007334 ***
Residuals  8  179.91   22.49                      

---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
===============================================================================
Least Significant Difference
===============================================================================

Study: model ~ "virus"

LSD t Test for yield 

Mean Square Error:  22.48917 

virus,  means and individual ( 95 %) CI

      yield      std r       LCL      UCL  Min  Max
cc 24.40000 3.609709 3 18.086268 30.71373 21.7 28.5
fc 12.86667 2.159475 3  6.552935 19.18040 10.6 14.9
ff 36.33333 7.333030 3 30.019601 42.64707 28.0 41.8
oo 36.90000 4.300000 3 30.586268 43.21373 32.1 40.4

alpha: 0.05 ; Df Error: 8
Critical Value of t: 2.306004 

Least Significant Difference 8.928965
Means with the same letter are not significantly different.

Groups, Treatments and means
a    oo      36.9 
a    ff      36.33 
b    cc      24.4 
c    fc      12.87 
===============================================================================
```

![crd-one-var-box-plots](https://cloud.githubusercontent.com/assets/276007/9181464/0ef70826-3f5a-11e5-98a9-459b908a72f7.png)
![crd-one-var-fit-plots](https://cloud.githubusercontent.com/assets/276007/9181469/134445ec-3f5a-11e5-925f-65b432b23da7.png)
![crd-one-var-bar-graph](https://cloud.githubusercontent.com/assets/276007/9181486/27e836e8-3f5a-11e5-8361-da17f6739d59.png)

moorepants committed:
This adds in example code for the single variable CRD example.

moorepants commented:
Oh yeah, you can view the code by clicking the "files changed" tab if you didn't know that.

moorepants commented:
I've added an RCBD example too. Please make sure the effects (interaction) plots are suitable.

lmpincus commented on examples/crd_one_var.R:
```diff
@@ -0,0 +1,66 @@
+# The ANOVA table shows that the virus factor is significant, so we then see
+# which levels are significant with respect to each other using least
+# significant difference.
+cat('Least Significant Differnce\n')
```
Difference

lmpincus commented on examples/rcbd_two_var.R:
```diff
@@ -0,0 +1,71 @@
+# so we then see which levels of both clone and nitrogen are significant with
+# respect to each other using least significant difference.
+# TODO : It isn't clear to me why the "block" variable wouldn't be checked here.
+cat('Least Significant Differnce\n')
```
Difference

moorepants committed:
Spelling.

moorepants committed:
Added seperators around the code that will be displayed to the user in the app.

moorepants commented:
@msimmond I added `#-------#` separators around the code that will be shown to the user.

moorepants committed:
Renamed the data frame to my_data.

moorepants committed:
Added Levene's Test.

moorepants committed:
Tidying up.

moorepants committed:
Updated the plots.

moorepants commented:
@msimmond I've updated this one with the types of plots you've requested on the others. I have some questions in the top comment box.

Commit:
Deleted 1-DF Tukey Test

Only valid when there's a blocking variable

msimmond commented on examples/crd_one_var.R:
```diff
@@ -0,0 +1,97 @@
+cat('Confidence Intervals\n')
+sep(79)
+#-----------------------------------------------------------------------------#
+confint(model)
```
I think we should remove the command confint(). The CI's that it produces don't match the ones in the LSD output.

moorepants replied:
But what does that mean? `confint` must give the correct confidence intervals for the base model. The LSD command may be doing something different. I'd hate to erase the correct one.

msimmond replied:
The LSD.test is giving 95% confidence intervals for the least square means of the 9 clone-nitrogen treatment combinations, which is what we want. The other confint() is not. 

For more background, a discussion that's related to this can be found here: http://stats.stackexchange.com/questions/29479/difference-between-ci-of-anova-level-coefficients-vs-t-test-ci-which-one-is-c

Commit:
create bar graph in ggplot

This is the type and characteristics of the bar graph that should be available for all scenarios.

Commit:
missing '+'

Commit:
made scale generic for bar plot

Commit:
increase font size

Commit:
increase font size

moorepants commented on examples/crd_one_var.R:
```diff
@@ -0,0 +1,138 @@
+merged_table <- merge(Data2,table1,by.x='virus',by.y='lsd.trt.names')
+
+#create bar graph
+library(ggplot2) 
```
A bar graph can be created without ggplot. Are you sure we you want to introduce ggplot based plots into the code that is displayed to the user? Seems like a lot of overhead to me. Creating ggplot plots use a very different API than simple R plots.

rkingdc replied:
It would be easy to create a very similar looking plot using base graphics (using barplot() and arrows() functions), reducing dependencies.

msimmond commented:
I haven't been noting whether code is displayed to user for anything so far. I'm not sure at this point whether the code should be displayed. But I do think these graphs are awesome. If you have something else that can do that, let me know.

msimmond commented:
This is one of the primary final figure results so I think they should have the LSD letter labels, and look really nice for reports.

moorepants commented on examples/crd_one_var.R:
```diff
@@ -0,0 +1,138 @@
+lsd.trt.names <- (lsd$groups[,1])
+table1 <- data.frame(lsd.trt.names,lsd.trt.means,lsd.letters)
+#Calc standard errors for error bars in graph, and save in temporary df 
+Data2 <- data.frame(summarySE(data=my.data, 
```
Seems like `summarySE` comes from a package called `Rmisc` which hasn't been loaded.

moorepants commented on examples/crd_one_var.R:
```diff
@@ -0,0 +1,138 @@
+lsd.letters <- as.character(lsd$groups[,3])
+lsd.trt.means <- (lsd$groups[,2])
+lsd.trt.names <- (lsd$groups[,1])
+table1 <- data.frame(lsd.trt.names,lsd.trt.means,lsd.letters)
```
`table1` is the exact same data frame as `lsd$groups`, there is no reason to create it like this.

moorepants commented:
> I haven't been noting whether code is displayed to user for anything so far. I'm not sure at this point whether the code should be displayed. But I do think these graphs are awesome. If you have something else that can do that, let me know.

> This is one of the primary final figure results so I think they should have the LSD letter labels, and look really nice for reports. 

I was under the impression that we are showing all of the code used to run the analysis and produce the graphs to the user so that they can learn to use R. If that it is the case, then having fancy graphics seems very secondary. I personally think we should keep the examples and results as simple as possible. If we were not showing the R code to the users, then we would be free to generate whatever results we want regardless of how complicated the code is.

I'm going to switch your final bar graph to something that uses builtin R calls instead of ggplot unless you disagree.

moorepants committed:
Cleaned up Maegan's code.

moorepants committed:
Fixed merge conflict.

msimmond commented:
It's not my understanding that the users are learning to use R.

If you can code it simpler for the bar graphs, please do. I just want them
to have all the same features.

On Mon, Aug 10, 2015 at 11:49 AM, Jason K. Moore <notifications@github.com>
wrote:

> I haven't been noting whether code is displayed to user for anything so
> far. I'm not sure at this point whether the code should be displayed. But I
> do think these graphs are awesome. If you have something else that can do
> that, let me know.
>
> This is one of the primary final figure results so I think they should
> have the LSD letter labels, and look really nice for reports.
>
> I was under the impression that we are showing all of the code used to run
> the analysis and produce the graphs to the user so that they can learn to
> use R. If that it is the case, then having fancy graphics seems very
> secondary. I personally think we should keep the examples and results as
> simple as possible. If we were not showing the R code to the users, then we
> would be free to generate whatever results we want regardless of how
> complicated the code is.
>
> I'm going to switch your final bar graph to something that uses builtin R
> calls instead of ggplot unless you disagree.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucd-ipo/aip-analysis/pull/33#issuecomment-129562935>.
>



-- 
Maegen Simmonds, Ph.D.
mbsimmonds@ucdavis.edu
707-694-6079

moorepants commented:
If the users are not learning are why are we displaying the R code to them and why is the R code in the report? If we aren't trying to teach them R then I can remove all the displayed code and it would simplify things greatly. I was told by both Kyle and Lauren that we were supposed to be designing it to be able to teach them R.

moorepants commented:
Is this one done? I've posted the current output and graphs at the top of this page. The only remaining issue is whether to simplify the bar chart. This bar chart should be able to be generated relatively easily with http://www.statmethods.net/graphs/bar.html. I'll come back to that once we figure out if we are showing R code or not.

moorepants committed:
Added png outputs.

moorepants commented:
Why was the Tukey test removed from this one? Which examples should the Tukey test be run on?

msimmond commented:
Tukey test for non-additivity is only done for RCBD with 1 replication per block (all our cases). It doesn't apply for a CRD.

moorepants commented:
Here is a potential example of a bar chart with the builtin R functions: http://monkeysuncle.stanford.edu/?p=485

moorepants commented:
@msimmond What's the verdict on whether users need to see the R code or not? If they don't need to see it, then the app code can be simplified a great deal.

moorepants committed:
Invisible.

moorepants commented:
All graph code will be hidden from the user.

moorepants commented:
This one is done. Merging.
